lib/Doctrine/ORM/Internal/Hydration/ObjectHydrator.php

```diff
@@ -281,9 +281,13 @@ private function getEntityFromIdentityMap(string $className, array $data)
             $idHash = '';
 
             foreach ($class->identifier as $fieldName) {
-                $idHash .= ' ' . (isset($class->associationMappings[$fieldName])
-                    ? $data[$class->associationMappings[$fieldName]['joinColumns'][0]['name']]
-                    : $data[$fieldName]);
+                if (isset($class->associationMappings[$fieldName])) {
+                    $idHash .= ' ' . $data[$class->associationMappings[$fieldName]['joinColumns'][0]['name']];
+                } elseif (class_exists('\BackedEnum') && is_a($data[$fieldName], '\BackedEnum')) {
```

Comment:
The `class_exists()` feels unnecessary and also wrong, given that `BackedEnum` is actually an interface.